packages/smooth_app/lib/pages/product/common/product_list_helper.dart

```diff
@@ -30,42 +31,35 @@ class ProductListHelper {
 
     addLine('Barcode', 'Name', 'Brand');
 
-    for (final String barcode in list.barcodes) {
-      if (barcode.length != 8 ||
-          barcode.length != 13 ||
-          int.tryParse(barcode) == null) {
-        continue;
-      }
+    final List<String> validBarcodes = list.barcodes.where(
+      (String barcode) {
+        return (barcode.length == 8 || barcode.length == 13) &&
```

Comment:
Nowhere else in the code we use those "8 or 13" length restrictions.
Is that relevant? Would it be useful in other cases?
Please put that in a static method anyway, easy to reuse, like "static bool isBarcodeClean(String)".

Thank you for the code rewriting btw.

Reply:
Of course! I've made it a static method

Reply:
I +1 @monsieurtanuki.
You already have this kind of method in the Dart SDK.

Your algorithm is also incorrect because barcodes may have leading zero(s).
In some cases, they are not provided in the String, but are still valid.